frontend/src/lib/components/LemonTable/TableRow.tsx

```diff
@@ -32,10 +32,13 @@ function TableRowRaw<T extends Record<string, any>>({
             <tr data-row-key={rowKeyDetermined} {...onRow?.(record)} className={rowClassNameDetermined}>
                 {!!expandable && rowExpandable >= 0 && (
                     <td>
-                        {rowExpandable && (
+                        {!!rowExpandable && (
                             <LemonButton
                                 type={isRowExpanded ? 'highlighted' : 'stealth'}
-                                onClick={() => setIsRowExpanded((state) => !state)}
+                                onClick={() => {
+                                    setIsRowExpanded((state) => !state)
+                                    !isRowExpanded && expandable.onRowExpanded && expandable.onRowExpanded(record)
```

Comment:
It may be a bit clearer if you just run that as a side effect of `setIsRowExpanded`.
```suggestion
                                    setIsRowExpanded((state) => {
                                        if (!state) {
                                            expandable.onRowExpanded?.(record)
                                        }
                                        return !state
                                    })
```

Reply:
that feels hidden to me, I'd expect `setIsRowExpanded` to not do anything besides setting that state